src/gz/bufread.rs

```diff
@@ -57,7 +57,7 @@ pub fn gz_encoder<R: BufRead>(header: Vec<u8>, r: R, lvl: Compression) -> GzEnco
     let crc = CrcReader::new(r);
     GzEncoder {
         inner: deflate::bufread::DeflateEncoder::new(crc, lvl),
-        header,
+        header: header.into_boxed_slice(),
```

Comment:
Generally I expect users to be more concerned by performance than storage size for these types. Converting a `Vec` into a boxed slice potentially reallocates, and I suspect the possible performance loss is not really worth the memory saving.

Reply:
Right! It's probably the [shedding of excess capacity](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.into_boxed_slice) that can cause [the reallocation](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.shrink_to_fit).

Let's keep the `Vec` then.